### PR TITLE
[Exec] Add specs for $0 and __FILE__ when loaded

### PR DESCRIPTION
**Pull request by segiddins**
Tests for #4488

### CODE REVIEW

File: spec/commands/exec_spec.rb [hunk 1]
```diff
@@ -457,5 +457,33 @@ def bin_path(a,b,c)
...
+
+      it_behaves_like "it runs"
+
+      context "when the path is relative" do
+        let(:path) { super().relative_path_from(Pathname.pwd) }
+
+        it_behaves_like "it runs"
+      end
+
+      xcontext "when the path is relative with a leading ./" do
```
Comment (segiddins) [inline spec/commands/exec_spec.rb]: note this is currently disabled since I haven't had time to dig into MRI to see if this is even possible

**Comment by indirect** [comment]
@segiddins rebase? I think we need this...

**Comment by segiddins** [comment]
Sure, but I don't actually know what the fix will be, these are just failing specs awaiting an implementation

**Comment by indirect** [comment]
do we have a way (maybe a flag to exec?) to opt out of exec using `load` for people who need $0 or __FILE__?

**Comment by segiddins** [comment]
At the moment we do not, but it should be easy to add a flag for it, `Bundler.settings[:disable_exec_load]` or something

**Comment by segiddins** [comment]
Rebase'd

**Comment by indirect** [comment]
@homu r+

**Pull request merged**